### PR TITLE
feat: Returns an error on join request with no display name.

### PR DESCRIPTION
**Pull request by damencho**
When someone tries to join a room with lobby enabled and display name is not set returns an error.

### CODE REVIEW

File: resources/prosody-plugins/mod_muc_lobby_rooms.lua
Comment (saghul) [inline resources/prosody-plugins/mod_muc_lobby_rooms.lua:403]: Shouldn't this be a separate module? This functionality is not exclusive to lobby is it?

Comment (damencho) [reply resources/prosody-plugins/mod_muc_lobby_rooms.lua:403]: This is for lobby, we are reporting this through jiconop when lobby is enabled so we can make the display name required.
https://github.com/jitsi/jitsi-meet/blob/e8766b265d7c7fe792335aa26bfd8bc644c3469f/resources/prosody-plugins/mod_muc_lobby_rooms.lua#L269

Now when dropping initial connection on opening pre-join we are adding this if someone tries to join without a display name when lobby is enabled to fail and set the UI that display name is required, so the participant enters a dispalyname and retries.

Comment (damencho) [reply resources/prosody-plugins/mod_muc_lobby_rooms.lua:403]: But this does not work if prejoin is not enabled ... the tests catch that failing case ...  need to look into

**Comment by hristoterezov** [comment]
@damencho What is the idea of the feature? Is it that only for lobby we require always a display name? Or do we want to always requre it?

I was wondering if we should add a config for enabling or disabling this?

**Comment by damencho** [comment]
@hristoterezov this is to implement a feature which we already have, but we will lose after merging the PR where we no longer create a connection on pre-join screen. 
And this is only for lobby, to require having a display name when lobby is enabled. And that feature and currently doesn't have any config to enable/disable it - we require display name in order to start knocking in Lobby.

**Comment by damencho** [comment]
The PR covers now the prejoin screen and not Lobby screen (this is when prejoin is disabled), so I'm working on adding that, which will require changes in both ljm and this PR. But I'll push everything on top of this, as a separate commits.

**Comment by hristoterezov** [comment]
Got it! Thanks for the explanation!

**Comment by damencho** [comment]
I have pushed the changes about the lobby screen when preJoin is disabled, which should fix the tests.

But I think we should not merge this till we merge https://github.com/jitsi/jitsi-meet/pull/13538 as there is logic here that assumes the connection is established when opening the prejoinscreen ... and we need to sync the changes from the other PR with this change and test it.

**Comment by damencho** [comment]
I force pushed as it is rebased now on top of https://github.com/jitsi/jitsi-meet/pull/13538 and is working with that change with no connection established while on prejoin screen.

**Review by hristoterezov — APPROVED** [review]

**Comment by damencho** [comment]
Jenkins test this please.

**Pull request merged**
